### PR TITLE
pkp/pkp-lib#3028 parse and save citations 

### PR DESCRIPTION
**Pull request by bozana**
s. https://github.com/pkp/pkp-lib/issues/3028

### CODE REVIEW

File: templates/controllers/modals/submissionMetadata/catalogEntryTabs.tpl [hunk 1]
```diff
@@ -34,7 +34,14 @@
...
 		</li>
-		{counter start=3 assign="counter"}
+		{assign var=tabsCount value=3}
+		{if $citationsEnabled}
+			<li>
+				<a name="citations" href="{url router=$smarty.const.ROUTE_COMPONENT component="tab.catalogEntry.CatalogEntryTabHandler" tab="citations" op="citations" submissionId=$submissionId stageId=$stageId tabPos="3"}">{translate key="submission.citations"}</a>
+			</li>
+			{assign var=tabsCount value=4}
+		{/if}
+		{counter start=$tabsCount assign="counter"}
```
Comment (NateWr) [inline templates/controllers/modals/submissionMetadata/catalogEntryTabs.tpl]: I think you can do this without adding `tabsCount`:

```diff
diff --git a/templates/controllers/modals/submissionMetadata/catalogEntryTabs.tpl b/templates/controllers/modals/submissionMetadata/catalogEntryTabs.tpl
index 56e1a69..f3fd26e 100644
--- a/templates/controllers/modals/submissionMetadata/catalogEntryTabs.tpl
+++ b/templates/controllers/modals/submissionMetadata/catalogEntryTabs.tpl
@@ -34,14 +34,13 @@
                <li>
                        <a name="catalog" href="{url router=$smarty.const.ROUTE_COMPONENT component="tab.catalogEntry.CatalogEntryTabHandler" tab="identifiers" op="identifiers" submissionId=$submissionId stageId=$stageId tabPos="2"}">{translate key="submission.identifiers"}</a>
                </li>
-               {assign var=tabsCount value=3}
+               {counter start=3 assign="counter"}
                {if $citationsEnabled}
                        <li>
-                               <a name="citations" href="{url router=$smarty.const.ROUTE_COMPONENT component="tab.catalogEntry.CatalogEntryTabHandler" tab="citations" op="citations" submissionId=$submissionId stageId=$stageId tabPos="3"}">{translate key="submission.citations"}</a>
+                               <a name="citations" href="{url router=$smarty.const.ROUTE_COMPONENT component="tab.catalogEntry.CatalogEntryTabHandler" tab="citations" op="citations" submissionId=$submissionId stageId=$stageId tabPos=$counter}">{translate key="submission.citations"}</a>
                        </li>
-                       {assign var=tabsCount value=4}
+                       {counter}
                {/if}
-               {counter start=$tabsCount assign="counter"}
                {call_hook name="Templates::Controllers::Modals::SubmissionMetadata::CatalogEntryTabs::Tabs"}
                {foreach from=$publicationFormats item=format}
                        <li>
```

File: templates/frontend/objects/monograph_full.tpl [hunk 2]
```diff
@@ -248,7 +248,18 @@
...
+					<h3 class="label">
+						{translate key="submission.citations"}
+					</h3>
+					<div class="value">
+						{iterate from=parsedCitations item=parsedCitation}
+							<p>{$parsedCitation->getRawCitation()|escape}</p>
+						{/iterate}
+					</div>
+				</div>
+			{elseif $monograph->getCitations()}
```
Comment (NateWr) [inline templates/frontend/objects/monograph_full.tpl]: Same thing here about moving the `if` into the `<div class="value">` element.

File: templates/frontend/objects/monograph_full.tpl [hunk 3]
```diff
@@ -248,13 +248,19 @@
...
 						{translate key="submission.citations"}
 					</h3>
 					<div class="value">
-						{$monograph->getCitations()|nl2br}
+						{if $parsedCitations->getCount()}
+							{iterate from=parsedCitations item=parsedCitation}
+								<p>{$parsedCitation->getRawCitation()|strip_unsafe_html}</p>
+							{/iterate}
+						{elseif $monograph->getCitations()}
+							{translate key="submission.citations"}
```
Comment (NateWr) [inline templates/frontend/objects/monograph_full.tpl]: Is this right? It doesn't look like it's printing the raw citations in this case.

Comment (bozana) [reply templates/frontend/objects/monograph_full.tpl]: au... :-O I removed/copied the wrong line... fix coming... THANKS!!!

**Pull request merged**